aten/src/ATen/native/README.md

```diff
@@ -346,6 +346,36 @@ added if applicable), so that it's still available for other backends to use.
 If you implemented a native function in C++ and want to find out which dispatch keyword
 should be used in native_functions.yaml, please [follow steps in dispatch keywords](#choosing-the-right-dispatch-keyword)
 
+### CompositeImplicitAutograd Compliance
+
+Functions registered as CompositeImplicitAutograd MUST work for most, if not
+all, backends. This means that we impose a set of constraints that make it more
+difficult to write a CompositeImplicitAutograd function than writing regular
+PyTorch code.
+
+If you wish to do something that is banned (you may wish to do this for perf
+reasons), please write a backwards formula for your operator so it is no longer
+CompositeImplicitAutograd or hide parts of the operator in a new operator
+that is not CompositeImplicitAutograd.
+
+CompositeImplicitAutograd operators must not:
+- call `set_` or `resize_` or moral equivalents. These are tricky to handle for
+many backends, like vmap and meta.
+- call `out=` operations. These are impossible to handle for vmap and can cause
+dispatch-to-python objects to lose their subclassing.
+- Change the metadata of a Tensor without performing dispatches. Examples of these
+operations are directly accessing the TensorImpl API to modify the
+sizes/strides/metadata of a Tensor.
+- In the same vein as the last point, `data_ptr` access or `item` access are not
```

Comment:
item should be fine no? It gets a plain number as output

Reply:
item isn't fine because meta tensors and wrapper Tensors choke on it

Reply:
To clarify a bit more... in C++ item() is literally just a call to .data_ptr and doesn't go through the dispatcher.

Reply:
ok